src/mono/mono/mini/interp/interp.c

```diff
@@ -1936,19 +1936,21 @@ interp_entry (InterpEntryData *data)
 
 	context->stack_pointer = (guchar*)sp;
 
-	g_assert (!context->has_resume_state);
-	g_assert (!context->safepoint_frame);
-
-	if (rmethod->needs_thread_attach)
-		mono_threads_detach_coop (orig_domain, &attach_cookie);
-
 	if (mono_llvm_only) {
-		if (context->has_resume_state)
-			mono_llvm_reraise_exception ((MonoException*)mono_gchandle_get_target_internal (context->exc_gchandle));
+		if (context->has_resume_state) {
+			MonoException *ex = (MonoException*)mono_gchandle_get_target_internal (context->exc_gchandle);
+			clear_resume_state (context);
+			mono_llvm_reraise_exception (ex);
```

Comment:
How can aot code catch this exception ? Also, how do we even resume to catch handler in aot only case, in normal operation ?

Reply:
This is more like a quick and dirty fix to avoid asserting, not sure this leads to correct behavior. The AOT code will catch this as a normal C++ exception.

Reply:
I don't understand how aot can catch C++ exceptions. Could you point me to the code that does this ?

Reply:
This is llvmonly mode, which uses llvm intrinsics to implement exception handling which is the same as c++ exception handling.

Reply:
Still investigating this, its possible that the assert is right, and this shouldn't happen.